src/RokuAdapter.ts

```diff
@@ -1128,43 +1165,52 @@ export class RequestPipeline {
                     this.emit('console-output', command);
                 }
                 this.client.write(commandText);
+                if (waitForPrompt) {
+                    this.isAtDebuggerPrompt = false;
+                }
             };
-            this.requests.push({
+
+            let request = {
                 executeCommand: executeCommand,
                 onComplete: (data) => {
                     console.debug(`Command finished (${waitForPrompt ? 'after waiting for prompt' : 'did not wait for prompt'}`, command);
                     console.debug('Data:', data);
                     resolve(data);
                 },
-                waitForPrompt: waitForPrompt
-            });
-            //start processing (safe to call multiple times)
-            this.process();
+                waitForPrompt: waitForPrompt,
+            };
+
+            if (!waitForPrompt) {
+                if (!this.isProcessing) {
```

Comment:
@chrisdp  There's no `else` condition for this...so there's a chance that `waitForPrompt === false && this.isProcessing === true`...that won't be handled by anything. That may cause issues.

Reply:
That is fine as this is what used to happen before due to the old check in the process function:
```
        if (this.isProcessing || this.requests.length === 0) {
            return;
        }
```

Reply:
But I guess it would just get processed later.
I think It might actutatly be find the way I have it to keep adding them to the que as only the wait for debugger ones will trigger processing with this change.